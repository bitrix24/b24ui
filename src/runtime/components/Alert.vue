<script lang="ts">
import type { AppConfig } from '@nuxt/schema'
import theme from '#build/b24ui/alert'
import type { AvatarProps, ButtonProps, IconComponent } from '../types'
import type { ComponentConfig } from '../types/utils'

type Alert = ComponentConfig<typeof theme, AppConfig, 'alert'>

export interface AlertProps {
  /**
   * The element or component this component should render as.
   * @defaultValue 'div'
   */
  as?: any
  title?: string
  description?: string
  /**
   * Display an icon on the left side.
   * @IconComponent
   */
  icon?: IconComponent
  avatar?: AvatarProps
  /**
   * @defaultValue 'default'
   */
  color?: Alert['variants']['color']
  /**
   * If set to `true` the color is inverted.
   * Used for 'air-primary', 'air-primary-success', 'air-primary-alert', 'air-primary-copilot' and 'air-primary-warning' colors.
   * @defaultValue false
   */
  inverted?: boolean
  /**
   * The orientation between the content and the actions.
   * @defaultValue 'vertical'
   */
  orientation?: Alert['variants']['orientation']
  /**
   * @defaultValue 'md'
   */
  size?: Alert['variants']['size']
  /**
   * Display a list of actions:
   * - under the title and description when orientation is `vertical`
   * - next to the close button when orientation is `horizontal`
   * `{ size: 'sm' }`{lang="ts"}
   */
  actions?: ButtonProps[]
  /**
   * Display a close button to dismiss the alert.
   * `{ size: 'sm', color: 'air-tertiary-no-accent' }`{lang="ts"}
   * @emits 'update:open'
   * @defaultValue false
   */
  close?: boolean | Partial<ButtonProps>
  /**
   * The icon displayed in the close button.
   * @defaultValue icons.close
   * @IconComponent
   */
  closeIcon?: IconComponent
  class?: any
  b24ui?: Alert['slots']
}

export interface AlertEmits {
  (e: 'update:open', value: boolean): void
}

export interface AlertSlots {
  leading(props?: {}): any
  title(props?: {}): any
  description(props?: {}): any
  actions(props?: {}): any
  close(props: { b24ui: { [K in keyof Required<Alert['slots']>]: (props?: Record<string, any>) => string } }): any
}
</script>

<script setup lang="ts">
import { computed } from 'vue'
import { Primitive } from 'reka-ui'
import { useAppConfig } from '#imports'
import { useLocale } from '../composables/useLocale'
import { tv } from '../utils/tv'
import icons from '../dictionary/icons'
import B24Avatar from './Avatar.vue'
import B24Button from './Button.vue'

const props = withDefaults(defineProps<AlertProps>(), {
  inverted: false,
  orientation: 'vertical'
})
const emits = defineEmits<AlertEmits>()
const slots = defineSlots<AlertSlots>()

const { t } = useLocale()
const appConfig = useAppConfig() as Alert['AppConfig']

const b24ui = computed(() => tv({ extend: tv(theme), ...(appConfig.b24ui?.alert || {}) })({
  color: props.color,
  inverted: Boolean(props.inverted),
  size: props.size,
  orientation: props.orientation,
  title: !!props.title || !!slots.title
}))
</script>

<template>
  <Primitive :as="as" :data-orientation="orientation" :class="b24ui.root({ class: [props.b24ui?.root, props.class] })">
    <slot name="leading">
      <Component
        :is="icon"
        v-if="icon"
        :class="b24ui.icon({ class: props.b24ui?.icon })"
      />
      <B24Avatar v-else-if="avatar" :size="((props.b24ui?.avatarSize || b24ui.avatarSize()) as AvatarProps['size'])" v-bind="avatar" :class="b24ui.avatar({ class: props.b24ui?.avatar })" />
    </slot>

    <div :class="b24ui.wrapper({ class: props.b24ui?.wrapper })">
      <div v-if="title || !!slots.title" :class="b24ui.title({ class: props.b24ui?.title })">
        <slot name="title">
          {{ title }}
        </slot>
      </div>
      <div v-if="description || !!slots.description" :class="b24ui.description({ class: props.b24ui?.description })">
        <slot name="description">
          {{ description }}
        </slot>
      </div>

      <div v-if="orientation === 'vertical' && (actions?.length || !!slots.actions)" :class="b24ui.actions({ class: props.b24ui?.actions })">
        <slot name="actions">
          <B24Button v-for="(action, index) in actions" :key="index" size="sm" v-bind="action" />
        </slot>
      </div>
    </div>

    <div v-if="(orientation === 'horizontal' && (actions?.length || !!slots.actions)) || close" :class="b24ui.actions({ class: props.b24ui?.actions, orientation: 'horizontal' })">
      <template v-if="orientation === 'horizontal' && (actions?.length || !!slots.actions)">
        <slot name="actions">
          <B24Button v-for="(action, index) in actions" :key="index" size="sm" v-bind="action" />
        </slot>
      </template>

      <slot name="close" :b24ui="b24ui">
        <B24Button
          v-if="close"
          :icon="closeIcon || icons.close"
          size="sm"
          color="air-tertiary"
          :aria-label="t('alert.close')"
          v-bind="(typeof close === 'object' ? close as Partial<ButtonProps> : {})"
          :class="b24ui.close({ class: props.b24ui?.close })"
          @click="emits('update:open', false)"
        />
      </slot>
    </div>
  </Primitive>
</template>
