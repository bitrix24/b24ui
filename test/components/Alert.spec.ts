import { describe, it, expect } from 'vitest'
import Alert from '../../src/runtime/components/Alert.vue'
import type { AlertProps, AlertSlots } from '../../src/runtime/components/Alert.vue'
import ComponentRender from '../component-render'
import SignIcon from '@bitrix24/b24icons-vue/main/SignIcon'
import Cross30Icon from '@bitrix24/b24icons-vue/actions/Cross30Icon'

describe('Alert', () => {
  const props = { title: 'Alert' }

  it.each([
    // Props
    ['with title', { props }],
    ['with description', { props: { ...props, description: 'Description' } }],
    ['with icon', { props: { ...props, icon: SignIcon } }],
    ['with avatar', { props: { ...props, avatar: { src: 'https://github.com/bitrix24.png' } } }],
    ['with actions', { props: { ...props, actions: [{ label: 'Action' }] } }],
    ['with orientation vertical', { props: { ...props, icon: SignIcon, description: 'This is a description', actions: [{ label: 'Action' }], orientation: 'vertical' as const } }],
    ['with orientation horizontal', { props: { ...props, icon: SignIcon, description: 'This is a description', actions: [{ label: 'Action' }], orientation: 'horizontal' as const } }],
    ['with close', { props: { ...props, close: true } }],
    ['with closeIcon', { props: { ...props, close: true, closeIcon: Cross30Icon } }],
    [`with success`, { props: { ...props, color: 'success' as const } }],
    ['with as', { props: { ...props, as: 'article' } }],
    ['with class', { props: { ...props, class: 'w-48' } }],
    ['with b24ui', { props: { ...props, b24ui: { title: 'font-bold' } } }],
    // Slots
    ['with leading slot', { props, slots: { title: () => 'Leading slot' } }],
    ['with title slot', { props, slots: { title: () => 'Title slot' } }],
    ['with description slot', { props, slots: { description: () => 'Description slot' } }],
    ['with close slot', { props, slots: { close: () => 'Close slot' } }]
  ])('renders %s correctly', async (nameOrHtml: string, options: { props?: AlertProps, slots?: Partial<AlertSlots> }) => {
    const html = await ComponentRender(nameOrHtml, options, Alert)
    expect(html).toMatchSnapshot()
  })
})
