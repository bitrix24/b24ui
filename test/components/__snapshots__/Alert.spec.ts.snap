// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Alert > renders with actions correctly 1`] = `
"<div data-orientation="vertical" class="relative overflow-hidden w-full rounded-3xs flex text-base-950 bg-base-200 dark:text-base-950 dark:bg-base-200 py-md ps-md pe-xs gap-2.5 items-start">
  <!--v-if-->
  <div class="min-w-0 flex-1 flex flex-col font-b24-primary font-normal">
    <div class="font-bold text-md leading-none">Alert</div>
    <!--v-if-->
    <div class="flex flex-wrap gap-1.5 shrink-0 items-start mt-2"><button type="button" class="select-none cursor-pointer inline-flex items-center relative outline-transparent focus-visible:outline-2 focus-visible:outline-offset-2 disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-50 aria-disabled:opacity-50 transition duration-150 ease-linear text-white dark:text-base-100 bg-base-650 border border-base-650 hover:bg-base-850 hover:border-base-850 hover:focus-visible:ring-0 active:bg-slate-850 active:border-slate-850 active:focus-visible:ring-0 disabled:bg-base-650 aria-disabled:bg-base-650 disabled:border-base-650 aria-disabled:border-base-650 focus-visible:outline-base-650 ring-base-850 focus:outline-none focus-visible:ring-base-850 h-xl2 ps-4 pe-4 text-3xs leading-none rounded-2xs font-bold uppercase">
        <!--v-if-->
        <div class="inline-flex items-center gap-1">
          <!--v-if--><span class="truncate">Action</span>
          <!--v-if-->
        </div>
      </button></div>
  </div>
  <!--v-if-->
</div>"
`;

exports[`Alert > renders with as correctly 1`] = `
"<article data-orientation="vertical" class="relative overflow-hidden w-full rounded-3xs flex text-base-950 bg-base-200 dark:text-base-950 dark:bg-base-200 py-md ps-md pe-xs gap-2.5 items-start">
  <!--v-if-->
  <div class="min-w-0 flex-1 flex flex-col font-b24-primary font-normal">
    <div class="font-bold text-md leading-none">Alert</div>
    <!--v-if-->
    <!--v-if-->
  </div>
  <!--v-if-->
</article>"
`;

exports[`Alert > renders with avatar correctly 1`] = `
"<div data-orientation="vertical" class="relative overflow-hidden w-full rounded-3xs flex text-base-950 bg-base-200 dark:text-base-950 dark:bg-base-200 py-md ps-md pe-xs gap-2.5 items-start"><span class="inline-flex items-center justify-center select-none rounded-full align-middle bg-base-100 dark:bg-base-900 size-12 text-2xl shrink-0"><img role="img" src="https://github.com/bitrix24.png" width="48" height="48" class="h-full w-full rounded-[inherit] object-cover"></span>
  <div class="min-w-0 flex-1 flex flex-col font-b24-primary font-normal">
    <div class="font-bold text-md leading-none">Alert</div>
    <!--v-if-->
    <!--v-if-->
  </div>
  <!--v-if-->
</div>"
`;

exports[`Alert > renders with b24ui correctly 1`] = `
"<div data-orientation="vertical" class="relative overflow-hidden w-full rounded-3xs flex text-base-950 bg-base-200 dark:text-base-950 dark:bg-base-200 py-md ps-md pe-xs gap-2.5 items-start">
  <!--v-if-->
  <div class="min-w-0 flex-1 flex flex-col font-b24-primary font-normal">
    <div class="text-md leading-none font-bold">Alert</div>
    <!--v-if-->
    <!--v-if-->
  </div>
  <!--v-if-->
</div>"
`;

exports[`Alert > renders with class correctly 1`] = `
"<div data-orientation="vertical" class="relative overflow-hidden rounded-3xs flex text-base-950 bg-base-200 dark:text-base-950 dark:bg-base-200 py-md ps-md pe-xs gap-2.5 items-start w-48">
  <!--v-if-->
  <div class="min-w-0 flex-1 flex flex-col font-b24-primary font-normal">
    <div class="font-bold text-md leading-none">Alert</div>
    <!--v-if-->
    <!--v-if-->
  </div>
  <!--v-if-->
</div>"
`;

exports[`Alert > renders with close correctly 1`] = `
"<div data-orientation="vertical" class="relative overflow-hidden w-full rounded-3xs flex text-base-950 bg-base-200 dark:text-base-950 dark:bg-base-200 py-md ps-md pe-xs gap-2.5 items-start">
  <!--v-if-->
  <div class="min-w-0 flex-1 flex flex-col font-b24-primary font-normal">
    <div class="font-bold text-md leading-none">Alert</div>
    <!--v-if-->
    <!--v-if-->
  </div>
  <div class="flex flex-wrap gap-1.5 shrink-0 items-center">
    <!--v-if--><button type="button" aria-label="Close" class="select-none cursor-pointer inline-flex items-center relative outline-transparent focus-visible:outline-2 focus-visible:outline-offset-2 disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-50 aria-disabled:opacity-50 transition duration-150 ease-linear no-underline decoration-solid decoration-auto decoration-gray-900 bg-transparent border border-transparent hover:focus-visible:ring-0 active:text-base-900 dark:active:text-base-700 active:focus-visible:ring-0 disabled:text-base-900 aria-disabled:text-base-900 dark:disabled:text-base-300 dark:aria-disabled:disabled:text-base-300 focus-visible:outline-base-400 ring-base-850 focus:outline-none focus-visible:ring-base-850 h-xl2 text-3xs leading-none rounded-2xs justify-center font-bold uppercase p-0 text-base-800 dark:text-base-800 hover:text-base-950 dark:hover:text-base-950">
      <!--v-if-->
      <div class="inline-flex items-center gap-1"><svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" aria-hidden="true" data-slot="icon" class="shrink-0 size-5">
          <path fill="currentColor" fill-rule="evenodd" d="m13.505 12 3.905 3.904-1.505 1.505L12 13.505l-3.905 3.904-1.505-1.505L10.495 12 6.59 8.095 8.095 6.59 12 10.495l3.905-3.905 1.505 1.505z" clip-rule="evenodd"></path>
        </svg>
        <!--v-if-->
        <!--v-if-->
      </div>
    </button>
  </div>
</div>"
`;

exports[`Alert > renders with close slot correctly 1`] = `
"<div data-orientation="vertical" class="relative overflow-hidden w-full rounded-3xs flex text-base-950 bg-base-200 dark:text-base-950 dark:bg-base-200 py-md ps-md pe-xs gap-2.5 items-start">
  <!--v-if-->
  <div class="min-w-0 flex-1 flex flex-col font-b24-primary font-normal">
    <div class="font-bold text-md leading-none">Alert</div>
    <!--v-if-->
    <!--v-if-->
  </div>
  <!--v-if-->
</div>"
`;

exports[`Alert > renders with closeIcon correctly 1`] = `
"<div data-orientation="vertical" class="relative overflow-hidden w-full rounded-3xs flex text-base-950 bg-base-200 dark:text-base-950 dark:bg-base-200 py-md ps-md pe-xs gap-2.5 items-start">
  <!--v-if-->
  <div class="min-w-0 flex-1 flex flex-col font-b24-primary font-normal">
    <div class="font-bold text-md leading-none">Alert</div>
    <!--v-if-->
    <!--v-if-->
  </div>
  <div class="flex flex-wrap gap-1.5 shrink-0 items-center">
    <!--v-if--><button type="button" aria-label="Close" class="select-none cursor-pointer inline-flex items-center relative outline-transparent focus-visible:outline-2 focus-visible:outline-offset-2 disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-50 aria-disabled:opacity-50 transition duration-150 ease-linear no-underline decoration-solid decoration-auto decoration-gray-900 bg-transparent border border-transparent hover:focus-visible:ring-0 active:text-base-900 dark:active:text-base-700 active:focus-visible:ring-0 disabled:text-base-900 aria-disabled:text-base-900 dark:disabled:text-base-300 dark:aria-disabled:disabled:text-base-300 focus-visible:outline-base-400 ring-base-850 focus:outline-none focus-visible:ring-base-850 h-xl2 text-3xs leading-none rounded-2xs justify-center font-bold uppercase p-0 text-base-800 dark:text-base-800 hover:text-base-950 dark:hover:text-base-950">
      <!--v-if-->
      <div class="inline-flex items-center gap-1"><svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" aria-hidden="true" data-slot="icon" class="shrink-0 size-5">
          <path fill="currentColor" fill-rule="evenodd" d="m13.505 12 3.905 3.904-1.505 1.505L12 13.505l-3.905 3.904-1.505-1.505L10.495 12 6.59 8.095 8.095 6.59 12 10.495l3.905-3.905 1.505 1.505z" clip-rule="evenodd"></path>
        </svg>
        <!--v-if-->
        <!--v-if-->
      </div>
    </button>
  </div>
</div>"
`;

exports[`Alert > renders with description correctly 1`] = `
"<div data-orientation="vertical" class="relative overflow-hidden w-full rounded-3xs flex text-base-950 bg-base-200 dark:text-base-950 dark:bg-base-200 py-md ps-md pe-xs gap-2.5 items-start">
  <!--v-if-->
  <div class="min-w-0 flex-1 flex flex-col font-b24-primary font-normal">
    <div class="font-bold text-md leading-none">Alert</div>
    <div class="text-md leading-[17px] mt-1">Description</div>
    <!--v-if-->
  </div>
  <!--v-if-->
</div>"
`;

exports[`Alert > renders with description slot correctly 1`] = `
"<div data-orientation="vertical" class="relative overflow-hidden w-full rounded-3xs flex text-base-950 bg-base-200 dark:text-base-950 dark:bg-base-200 py-md ps-md pe-xs gap-2.5 items-start">
  <!--v-if-->
  <div class="min-w-0 flex-1 flex flex-col font-b24-primary font-normal">
    <div class="font-bold text-md leading-none">Alert</div>
    <div class="text-md leading-[17px] mt-1">Description slot</div>
    <!--v-if-->
  </div>
  <!--v-if-->
</div>"
`;

exports[`Alert > renders with icon correctly 1`] = `
"<div data-orientation="vertical" class="relative overflow-hidden w-full rounded-3xs flex text-base-950 bg-base-200 dark:text-base-950 dark:bg-base-200 py-md ps-md pe-xs gap-2.5 items-start"><svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" aria-hidden="true" data-slot="icon" class="shrink-0 size-6">
    <path fill="currentColor" fill-rule="evenodd" d="M11.985 3a1.706 1.706 0 0 0-1.699 1.825L6.774 7.304h-1.7a1 1 0 0 0-1 1v9.203a1 1 0 0 0 1 1h13.822a1 1 0 0 0 1-1V8.304a1 1 0 0 0-1-1H17.19l-3.51-2.447q.006-.075.007-.15c0-.94-.763-1.707-1.703-1.707M10.8 5.931a1.694 1.694 0 0 0 2.35.019l1.943 1.354H8.855zm-4.231 4.744a.8.8 0 0 1 .8-.8h7.115a.8.8 0 0 1 .8.8v.137a.8.8 0 0 1-.8.8H7.37a.8.8 0 0 1-.8-.8zm.8 2.845a.8.8 0 0 0-.8.8v.137a.8.8 0 0 0 .8.8H13.1a.8.8 0 0 0 .8-.8v-.138a.8.8 0 0 0-.8-.8z" clip-rule="evenodd"></path>
  </svg>
  <div class="min-w-0 flex-1 flex flex-col font-b24-primary font-normal">
    <div class="font-bold text-md leading-none">Alert</div>
    <!--v-if-->
    <!--v-if-->
  </div>
  <!--v-if-->
</div>"
`;

exports[`Alert > renders with leading slot correctly 1`] = `
"<div data-orientation="vertical" class="relative overflow-hidden w-full rounded-3xs flex text-base-950 bg-base-200 dark:text-base-950 dark:bg-base-200 py-md ps-md pe-xs gap-2.5 items-start">
  <!--v-if-->
  <div class="min-w-0 flex-1 flex flex-col font-b24-primary font-normal">
    <div class="font-bold text-md leading-none">Leading slot</div>
    <!--v-if-->
    <!--v-if-->
  </div>
  <!--v-if-->
</div>"
`;

exports[`Alert > renders with orientation horizontal correctly 1`] = `
"<div data-orientation="horizontal" class="relative overflow-hidden w-full rounded-3xs flex text-base-950 bg-base-200 dark:text-base-950 dark:bg-base-200 py-md ps-md pe-xs gap-2.5 items-center"><svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" aria-hidden="true" data-slot="icon" class="shrink-0 size-6">
    <path fill="currentColor" fill-rule="evenodd" d="M11.985 3a1.706 1.706 0 0 0-1.699 1.825L6.774 7.304h-1.7a1 1 0 0 0-1 1v9.203a1 1 0 0 0 1 1h13.822a1 1 0 0 0 1-1V8.304a1 1 0 0 0-1-1H17.19l-3.51-2.447q.006-.075.007-.15c0-.94-.763-1.707-1.703-1.707M10.8 5.931a1.694 1.694 0 0 0 2.35.019l1.943 1.354H8.855zm-4.231 4.744a.8.8 0 0 1 .8-.8h7.115a.8.8 0 0 1 .8.8v.137a.8.8 0 0 1-.8.8H7.37a.8.8 0 0 1-.8-.8zm.8 2.845a.8.8 0 0 0-.8.8v.137a.8.8 0 0 0 .8.8H13.1a.8.8 0 0 0 .8-.8v-.138a.8.8 0 0 0-.8-.8z" clip-rule="evenodd"></path>
  </svg>
  <div class="min-w-0 flex-1 flex flex-col font-b24-primary font-normal">
    <div class="font-bold text-md leading-none">Alert</div>
    <div class="text-md leading-[17px] mt-1">This is a description</div>
    <!--v-if-->
  </div>
  <div class="flex flex-wrap gap-1.5 shrink-0 items-center"><button type="button" class="select-none cursor-pointer inline-flex items-center relative outline-transparent focus-visible:outline-2 focus-visible:outline-offset-2 disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-50 aria-disabled:opacity-50 transition duration-150 ease-linear text-white dark:text-base-100 bg-base-650 border border-base-650 hover:bg-base-850 hover:border-base-850 hover:focus-visible:ring-0 active:bg-slate-850 active:border-slate-850 active:focus-visible:ring-0 disabled:bg-base-650 aria-disabled:bg-base-650 disabled:border-base-650 aria-disabled:border-base-650 focus-visible:outline-base-650 ring-base-850 focus:outline-none focus-visible:ring-base-850 h-xl2 ps-4 pe-4 text-3xs leading-none rounded-2xs font-bold uppercase">
      <!--v-if-->
      <div class="inline-flex items-center gap-1">
        <!--v-if--><span class="truncate">Action</span>
        <!--v-if-->
      </div>
    </button>
    <!--v-if-->
  </div>
</div>"
`;

exports[`Alert > renders with orientation vertical correctly 1`] = `
"<div data-orientation="vertical" class="relative overflow-hidden w-full rounded-3xs flex text-base-950 bg-base-200 dark:text-base-950 dark:bg-base-200 py-md ps-md pe-xs gap-2.5 items-start"><svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" aria-hidden="true" data-slot="icon" class="shrink-0 size-6">
    <path fill="currentColor" fill-rule="evenodd" d="M11.985 3a1.706 1.706 0 0 0-1.699 1.825L6.774 7.304h-1.7a1 1 0 0 0-1 1v9.203a1 1 0 0 0 1 1h13.822a1 1 0 0 0 1-1V8.304a1 1 0 0 0-1-1H17.19l-3.51-2.447q.006-.075.007-.15c0-.94-.763-1.707-1.703-1.707M10.8 5.931a1.694 1.694 0 0 0 2.35.019l1.943 1.354H8.855zm-4.231 4.744a.8.8 0 0 1 .8-.8h7.115a.8.8 0 0 1 .8.8v.137a.8.8 0 0 1-.8.8H7.37a.8.8 0 0 1-.8-.8zm.8 2.845a.8.8 0 0 0-.8.8v.137a.8.8 0 0 0 .8.8H13.1a.8.8 0 0 0 .8-.8v-.138a.8.8 0 0 0-.8-.8z" clip-rule="evenodd"></path>
  </svg>
  <div class="min-w-0 flex-1 flex flex-col font-b24-primary font-normal">
    <div class="font-bold text-md leading-none">Alert</div>
    <div class="text-md leading-[17px] mt-1">This is a description</div>
    <div class="flex flex-wrap gap-1.5 shrink-0 items-start mt-2"><button type="button" class="select-none cursor-pointer inline-flex items-center relative outline-transparent focus-visible:outline-2 focus-visible:outline-offset-2 disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-50 aria-disabled:opacity-50 transition duration-150 ease-linear text-white dark:text-base-100 bg-base-650 border border-base-650 hover:bg-base-850 hover:border-base-850 hover:focus-visible:ring-0 active:bg-slate-850 active:border-slate-850 active:focus-visible:ring-0 disabled:bg-base-650 aria-disabled:bg-base-650 disabled:border-base-650 aria-disabled:border-base-650 focus-visible:outline-base-650 ring-base-850 focus:outline-none focus-visible:ring-base-850 h-xl2 ps-4 pe-4 text-3xs leading-none rounded-2xs font-bold uppercase">
        <!--v-if-->
        <div class="inline-flex items-center gap-1">
          <!--v-if--><span class="truncate">Action</span>
          <!--v-if-->
        </div>
      </button></div>
  </div>
  <!--v-if-->
</div>"
`;

exports[`Alert > renders with success correctly 1`] = `
"<div data-orientation="vertical" class="relative overflow-hidden w-full rounded-3xs flex text-green-780 bg-green-200 dark:text-green-780 dark:bg-green-300 py-md ps-md pe-xs gap-2.5 items-start">
  <!--v-if-->
  <div class="min-w-0 flex-1 flex flex-col font-b24-primary font-normal">
    <div class="font-bold text-md leading-none">Alert</div>
    <!--v-if-->
    <!--v-if-->
  </div>
  <!--v-if-->
</div>"
`;

exports[`Alert > renders with title correctly 1`] = `
"<div data-orientation="vertical" class="relative overflow-hidden w-full rounded-3xs flex text-base-950 bg-base-200 dark:text-base-950 dark:bg-base-200 py-md ps-md pe-xs gap-2.5 items-start">
  <!--v-if-->
  <div class="min-w-0 flex-1 flex flex-col font-b24-primary font-normal">
    <div class="font-bold text-md leading-none">Alert</div>
    <!--v-if-->
    <!--v-if-->
  </div>
  <!--v-if-->
</div>"
`;

exports[`Alert > renders with title slot correctly 1`] = `
"<div data-orientation="vertical" class="relative overflow-hidden w-full rounded-3xs flex text-base-950 bg-base-200 dark:text-base-950 dark:bg-base-200 py-md ps-md pe-xs gap-2.5 items-start">
  <!--v-if-->
  <div class="min-w-0 flex-1 flex flex-col font-b24-primary font-normal">
    <div class="font-bold text-md leading-none">Title slot</div>
    <!--v-if-->
    <!--v-if-->
  </div>
  <!--v-if-->
</div>"
`;
